### ISSUE TITLE
Is it possible to get a cidr range for IP whitelisting?

### ISSUE THREAD

**Issue by justinmchase** [denoland/deploy_feedback#289]
So suppose I provision an Azure Cosmos DB in my azure subscription. I then take the connection string and put it into a Deno secret and my Deno Deploy servers can now connect to the database, thats great. 

But since this is all in the cloud basically everyone on the internet can also connect to my database directly, though they don't have a valid connection string so its _mostly_ secured.

However, the next level of security, to more properly secure this database, would be to add some IP rules to the database to prevent anything other than my servers from accessing the database at all. Normally I'd do this by adding a nat gateway to the VNET running my servers and hooking it up to an IP prefix which forces all network traffic to egress out through one of the ip's in that prefix. I could then add the CIDR rule for that ip prefix to the database IP filter rules and then that would be properly secured.

Additionally, it would be important to not only have a nat gateway but to have one that is scoped specifically just for a single deploy application. So a hacker wouldn't be able to just go and create their own Deno deploy application to bypass my CIDR restrictions. I could see it being scoped to an org maybe but also to an individual application would probably be necessary.

So my question is just if there is a way to get the CIDR for network egress already? If not then could that be possible feature request? I could see this being an advanced for-paying-customers-only feature but I can say lacking this feature will be prohibitive for corporate adoption based purely on security requirements.

**Comment by justinmchase** [comment]
tldr;

What is the CIDR range for deno deploy egress?

**Comment by justinmchase** [comment]
Bueller?

**Comment by lucacasonato** [comment]
Deno Deploy does not use a fixed CIDR range for egress. Egress IPs vary drastically during the day.

Additionally, in a multi tenant cloud system like Deno Deploy, you really do not want to be doing access control based on egress IPs, because the egress IPs are shared by many tenants. For example, imagine that you limit access to your database by allow-listing certain IPs, but then a second Deno Deploy customer spins up a project that accesses that same database - they will use the same IP addresses (ie addresses are not unique to the customer / project).

Allow listing all Deno Deploy IPs for access control would be about as useful as allow-listing all AWS or GCP IPs: not very useful.

We are currently not planning to provide per-customer egress IPs.

**Comment by justinmchase** [comment]
> Allow listing all Deno Deploy IPs for access control would be about as useful as allow-listing all AWS or GCP IPs: not very useful.

Not quite, it would be a lot less broad than all of AWS or GCP, an attacker would have to know that it was in the Deploy IP range and it they would have to stage their attack from Deploy, which is quite a bit more useful then allowing no ip ranges at all.

> We are currently not planning to provide per-customer egress IPs.

Just saying, this will become a problem for corporate customers. This will be a 100% required feature by policy where I work, or we won't even be able to think about it. Seems like a checkbox on the premium plan feature list really. I'm imagining the tenant just gets a whole cluster, domain, VPC, etc. and doesn't share it with anyone else and just pays appropriately. You have one lower tier multi-tenant and then multiple single-tenant instances. Worth considering!

Also worth noting, Vercel also has this same limitation:
https://vercel.com/guides/how-to-allowlist-deployment-ip-address

I would think this would be a major feature advantage you could put on the sales form.
